packages/fleather/lib/src/widgets/editor.dart

```diff
@@ -2552,11 +2552,22 @@ class _UpdateTextSelectionVerticallyAction<
             : intent.forward
                 ? currentRun.moveNext()
                 : currentRun.movePrevious();
-    final TextPosition newExtent = shouldMove
-        ? currentRun.current
-        : intent.forward
-            ? TextPosition(offset: state.textEditingValue.text.length)
-            : const TextPosition(offset: 0);
+
+    TextPosition computeNewExtent() {
+      if (shouldMove) return currentRun.current;
+
+      if (intent.forward) {
+        if (collapseSelection) {
```

Comment:
Why are we conditioning call to `updateLastKnownWithSelection` to `collapseSelection`? Wouldn't it cause the same problem in situations where selection is supposed to extend?

Reply:
Because we don't want to set a collapses selection on remote if it is not.
In case of extended selection, there is no need for this I think